### PR TITLE
Skip sending the proxyReq event when the expect header is present

### PR DESCRIPTION
**Pull request by jsmylnycky**
Hotfix for https://www.npmjs.com/advisories/1486

Expecting build error due to Node 6.  Waiting for https://github.com/http-party/node-http-proxy/pull/1397  to be merged to have a clean CI build.

### CODE REVIEW

File: test/lib-http-proxy-passes-web-incoming-test.js [hunk 1]
```diff
@@ -126,6 +126,50 @@ describe('#createProxyServer.web() using own http server', function () {
...
+      source.close();
+      proxyServer.close();
+      expect(req.headers['x-special-proxy-header']).to.not.eql('foobar');
+      done();
+    });
+
+    proxyServer.listen('8081');
+    source.listen('8080');
+
+    const postData = ''.padStart(10025, 'x');
```
Comment (jkyto) [inline test/lib-http-proxy-passes-web-incoming-test.js:154]: 10025 or 1025? Does not make any difference but to align with the advisory.

**Comment by jimmyandrade** [comment]
+1

**Comment by alexgvozden** [comment]
will anyone merge this if it solves the issue?

**Comment by amitmula** [comment]
Any ETA on when this is getting merged ?

**Comment by Hypnosphi** [comment]
@indexzero @jcrugzz looks like this requires your immediate attention

**Comment by indexzero** [comment]
My children are the only thing that requires immediate attention, sorry. Software happens during normal working hours. Didn't get to this on Friday, therefore it will be tomorrow.

Jarrett may have a moment, I have asked him.

**Comment by jcrugzz** [comment]
@jsmylnycky thanks for the work here. Will release this fix in a few

**Comment by jcrugzz** [comment]
published as `1.18.1`

**Comment by fabb** [comment]
Have you informed npm support to whitelist this version? Currently it‘s still blacklisted: https://www.npmjs.com/advisories/1486/versions
The support usually resolves such inquiries within a few hours: security@npmjs.com

**Comment by Hypnosphi** [comment]
@indexzero that's understandable, sorry for my wording. But the [vulnerability](https://www.npmjs.com/advisories/1486) seems reported almost 3 months ago. Do you consider adding more core maintainers as an option?

**Comment by jsmylnycky** [comment]
@Hypnosphi If you take a look at the top of the Issues page, there's two pinned posts going back to Aug/Sept, basically looking to get more people active with the future of this project.  There's been very little activity from folks willing to actually jump in and contribute tho.  If it is something you're interested in doing, I suggest you take a look at those posts and leave some comments to get in touch :)

**Comment by Lucidiot** [comment]
Just out of curiosity, was the vulnerability *actually* reported to the maintainers? This would not be the first time nobody knows about the issue until the advisory goes public: https://github.com/sass/node-sass/issues/2816#issuecomment-574434620

**Comment by hendrikmolder** [comment]
> Have you informed npm support to whitelist this version? Currently it‘s still blacklisted: https://www.npmjs.com/advisories/1486/versions
> The support usually resolves such inquiries within a few hours: [security@npmjs.com](mailto:security@npmjs.com)

They've now marked the fixed version as unaffected

**Comment by nutza249949** [comment]
k

**Review by ryan-ally — COMMENTED** [review]
good

**Review by calbearox — COMMENTED** [review]
//
// Create an HTTP proxy server with an HTTPS target
//
httpProxy.createProxyServer({
  target: {
    protocol: 'https:',
    host: 'my-domain-name',
    port: 443,
    pfx: fs.readFileSync('path/to/certificate.p12'),
    passphrase: 'password',
  },
  changeOrigin: true,
}).listen(8000);

**Pull request merged**